packages/knip/docs/lsp-mode.md

```diff
@@ -0,0 +1,177 @@
+# Knip Language Server Protocol (LSP) Mode
+
+Knip now supports running as a Language Server Protocol (LSP) server, enabling real-time integration with any LSP-compatible editor (VS Code, Neovim, Emacs, etc.).
+
+## Features
+
+- **Real-time diagnostics**: See Knip issues directly in your editor as you code
+- **Code actions**: Quick fixes for common issues (remove unused exports, add missing dependencies)
+- **File watching**: Automatic re-analysis when configuration files change
+- **Workspace support**: Handle multiple workspace folders
+- **Configurable settings**: Control what types of issues are reported
+
+## Usage
+
+### As a standalone LSP server
+
+```bash
+# Run Knip in LSP mode
+knip --lsp
+
+# Or use the dedicated binary
+knip-lsp
+```
+
+### As a reporter
+
+```bash
+# Output LSP-formatted diagnostics
+knip --reporter lsp
```

Comment:
There is no such reporter?

Reply:
yea, i started out trying to make it a reporter, but that imposes certain restrictions on how it runs (LSP reruns things on change). I thought i removed all refs, will clean this up